gremlin-go/README.md

```diff
@@ -117,47 +68,114 @@ require (
 ```
 As well as a populated `go.sum` file.
 
-This following example should run, provided that it is configured to point to a compatible `gremlin-server`. In this example, a simple local server is running on port 8182, and this will print`[2]` as an output. If no server is available, this code can still be executed to print an error as output.
+*if there are no usages for gremlingo found, it will remove the require from go.mod and not import dependencies.*
+
+## Simple usage
 
+Here is a simple example of using Gremlin-Go as an import in a sample project's `main.go` file. 
 ```go
 package main
 
 import (
-	"fmt"
 	"github.com/apache/tinkerpop/gremlin-go/driver"
 )
 
+func main() {
+	// Simple stub to use the import. See subsequent section for actual usage. 
+	_, _ = gremlingo.NewDriverRemoteConnection(fmt.Sprintf("ws://%s:%d/gremlin", "localhost", 8182))
+}
+```
+
+*Pay attention to the suffix `/gremlin` in connection string*
+
+This following example should run, provided that it is configured to point to a compatible `gremlin-server`. In this example, a simple local server is running on port 8182, and this will print `[0]` as an output. If no server is available, this code can still be executed to print an error as output.
+
+```go
+package main
+
+import (
+	"fmt"
+	"github.com/apache/tinkerpop/gremlin-go/v3/driver"
+)
+
 func main() {
 	// Creating the connection to the server.
-	driverRemoteConnection, err := gremlingo.NewDriverRemoteConnection(fmt.Sprintf("ws://%s:%d", "localhost", 8182))
```

Comment:
Why remove fmt.Sprintf? Other example uses fmt.Sprintf.

Reply:
I'm not sure I like having Sprintf at all in our docs.

It's not immediately intuitive what it does unless you are particularly familiar with Go.

Reply:
I think in real life connection string is taken from app configuration.